src/hooks/useLayoutPrefs.ts

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+
+import { usePreferencesContext } from '@/contexts/PreferencesContext';
+import logger from '@/logger';
+
+/**
+ * Custom hook that provides storage interface for react-resizable-panels
+ * with debounced updates to reduce API calls when resizing panels
+ * See example implementation here: https://react-resizable-panels.vercel.app/examples/external-persistence
+ * Note that the custom storage interface must expose a synchronous getItem and setItem method.
+ */
+
+const DEBOUNCE_MS = 500;
+
+// Layout keys for the two possible panel combinations
+const LAYOUT_KEYS_WITH_PROPERTIES = 'main, properties, sidebar';
+const LAYOUT_KEYS_WITHOUT_PROPERTIES = 'main, sidebar';
+
+export default function useLayoutPrefs() {
+  const [showPropertiesDrawer, setShowPropertiesDrawer] =
+    React.useState<boolean>(false);
+  const { layout, handleUpdateLayout } = usePreferencesContext();
+
+  const timerRef = React.useRef<number | null>(null);
+
+  const debouncedUpdateLayout = React.useCallback(
+    (newLayout: string) => {
+      if (timerRef.current !== null) {
+        window.clearTimeout(timerRef.current);
+      }
+
+      timerRef.current = window.setTimeout(() => {
+        handleUpdateLayout(newLayout).catch(error => {
+          logger.debug('Failed to update layout:', error);
+        });
+        timerRef.current = null;
+      }, DEBOUNCE_MS);
+    },
+    [handleUpdateLayout]
+  );
+
+  const togglePropertiesDrawer = () => {
+    setShowPropertiesDrawer(prev => !prev);
+  };
+
+  // Initialize layouts from saved preferences
+  React.useEffect(() => {
```

Comment:
Without looking too deeply, my guess is a useEffect is not triggering when it should, causing the properties drawer state to be set incorrectly.